internal/controller/stas/ssa_client.go

```diff
@@ -76,6 +78,21 @@ func (fieldValidationStrict) ApplyToSubResourcePatch(opts *client.SubResourcePat
 	opts.Raw.FieldValidation = "Strict"
 }
 
+func NewConditionsPatch(existingConditions []metav1.Condition, conditions ...*metav1ac.ConditionApplyConfiguration) []*metav1ac.ConditionApplyConfiguration {
+	for _, condition := range conditions {
+		if condition.LastTransitionTime.IsZero() {
+			existingCondition := meta.FindStatusCondition(existingConditions, *condition.Type)
+			if existingCondition != nil && existingCondition.Status == *condition.Status {
+				condition.WithLastTransitionTime(existingCondition.LastTransitionTime)
+			} else {
+				condition.WithLastTransitionTime(metav1.NewTime(time.Now()))
```

Comment:
Don't think we'll ever create multiple conditions now, but if we would, I think it would make sense for all of them to have the same transition time. Meaning `time.Now()` should probably be moved outside of the loop?

Reply:
That would mean we risk creating something that never will be used. And it is not done where we are coming from: https://github.com/kubernetes/apimachinery/blob/master/pkg/api/meta/conditions.go#L49